Content.Server/Shuttles/Commands/FTLDiskBurnerCommand.cs

```diff
@@ -0,0 +1,182 @@
+/// </summary>
+[AdminCommand(AdminFlags.Fun)]
+
+public sealed class FTLDiskBurnerCommand : LocalizedCommands
+{
+    [Dependency] private readonly IEntityManager _entManager = default!;
+    [Dependency] private readonly IEntitySystemManager _entSystemManager = default!;
+
+    public override string Command => "FTLdiskburner";
+
+    [ValidatePrototypeId<EntityPrototype>]
+    public const string CoordinatesDisk = "CoordinatesDisk";
+    public override void Execute(IConsoleShell shell, string argStr, string[] args)
+    {
+        if (args.Length == 0)
+        {
+            shell.WriteError("Not enough arguments.");
+            return;
+        }
+
+
+        var player = shell.Player;
+
+        if (player == null)
+        {
+            shell.WriteLine("Only a player can run this command.");
+            return;
+        }
+
+        if (player.AttachedEntity == null)
+        {
+            shell.WriteLine("You don't have an entity to spawn a disk at.");
+            return;
+        }
+
+        EntityUid entity = player.AttachedEntity.Value;
+        var coords = _entManager.GetComponent<TransformComponent>(entity).Coordinates;
+
+        foreach (var destinations in args)
+        {
+            if (destinations == null)
+            {
+                shell.WriteLine("Destinations returned as null!");
+                DebugTools.AssertNotNull(destinations);
+                return;
+            }
+
+            // make sure destination is an id.
+            EntityUid dest;
+            if (_entManager.TryParseNetEntity(destinations, out var nullableDest))
+            {
+                if (nullableDest == null)
+                    continue;
+
+                dest = (EntityUid) nullableDest;
+
+                // we need to go to a map, so check if the EntID is something else then try for its map
+                if (!_entManager.HasComponent<MapComponent>(dest))
+                {
+                    if (!_entManager.TryGetComponent<TransformComponent>(dest, out var entTransform))
+                    {
+                        shell.WriteLine(destinations + " has no Transform!");
+                        continue;
+                    }
+
+                    var mapSystem = _entSystemManager.GetEntitySystem<SharedMapSystem>();
+                    mapSystem.TryGetMap(entTransform.MapID, out var mapDest);
+                    if (mapDest == null)
+                    {
+                        shell.WriteLine(destinations + " has no map to FTL to!");
+                        continue;
+                    }
+
+                    dest = (EntityUid) mapDest;
+                }
+
+                // find and verify the map is not somehow unusable.
+                if (!_entManager.TryGetComponent<MapComponent>(dest, out var mapComp))
+                {
+                    shell.WriteLine(destinations + " is somehow on map " + dest + " with no map component. What the fuck.");
+                    continue;
+                }
```

Comment:
Unfortunately we're subtly changing from checking any entity in the previous if, to checking what should only be a map in this one. Trying to squeeze it into the previous one would I think mean giving up on handling non-map entities by finding their map.

Or to put it another way, it will need to check whether dest is a map twice regardless since it changes what dest is (from any random entity to hopefully a map entity). Not nesting the if just handles scoping without defining it beforehand.

Reply:
Well, hold on maybe I can put all three of the later checks in...

Reply:
hmmm.... I suppose TryGetMap would probably return the map anyway if you passed it a map entity. I'm not sure that saves us anything though since (ideally) this whole blocks wont be running 95% of the time anyway since it tries to autocomplete a map entity in the helper.

Its probably just a smidge better to keep it the way it is though since doing that would mean running the TryGetComponent<TransformComponent> and TryGetMap calls 100% of the time when they're unlikely to be needed.